src/ecwrapper.cpp

```diff
@@ -13,6 +13,28 @@
 
 namespace {
 
+class ecgroup_wrapper
+{
+public:
+    ecgroup_wrapper()
+    : pgroup(EC_GROUP_new_by_curve_name(NID_secp256k1)){}
+
+    ~ecgroup_wrapper()
+    {
+        EC_GROUP_free(pgroup);
+    }
+
+    inline const EC_GROUP* get() const
+    {
+        return pgroup;
+    }
+
+private:
+    EC_GROUP* pgroup;
+};
+
+static const ecgroup_wrapper ecgroup_instance;
```

Comment:
Is it ok that this static initialization happens before 'official' OpenSSL initialization, even before main()?
What will happen if secp256k1 is not built into OpenSSL? I don't think it will reach the sanity check anymore.